cpp/src/shuffler/chunk.cpp

```diff
@@ -356,8 +363,14 @@ Chunk Chunk::concat(
         // Process data
         if (chunk.is_data_buffer_set() && chunk.concat_data_size() > 0) {
             // Copy data
-            std::ignore = chunk.data_->copy_to(
-                *concat_data, std::ptrdiff_t(curr_data_offset), stream, false
+            buffer_copy(
+                *concat_data,
+                *chunk.data_,
+                chunk.data_->size,
+                std::ptrdiff_t(curr_data_offset),  // dst_offset
+                0,  // src_offset
+                stream,
+                false
```

Comment:
Just to be sure, this is `false` here because in line 394 below (new line number) the event is overridden, is that correct?